### ISSUE TITLE
Confusion in default_currency type

### ISSUE THREAD

**Issue by washeck** [django-money/django-money#490]
There seems to be a confusion in how we understand default currency. The README says you can set it to None like this
`MoneyField(max_digits=10, decimal_places=2, null=True, default_currency=None)`

Unfortunately, this does not work for me because `MoneyField.deconstruct()` has the following code:
```
        if self.default_currency != DEFAULT_CURRENCY:
            kwargs['default_currency'] = str(self.default_currency)
```
which leads to funny situation when you end up with string 'None' in the db.

DEFAULT_CURRENCY is not documented in README but there is a comment in `settings.py`
```
# The default currency, you can define this in your project's settings module
# This has to be a currency object imported from moneyed
DEFAULT_CURRENCY = getattr(settings, 'DEFAULT_CURRENCY', DEFAULT_CURRENCY)
```

I find it confusing to have DEFAULT_CURRENCY in settings as object and default_currency in kwargs as string. I can submit PR but I am not sure what approach you prefer. As a minimal fix I suggest to special-case None in the kwargs and to add asserts for expected types (str or Currency).

**Comment by benjaoming** [comment]
Sounds like a bug. @Stranger6667 should we implement a fix that follows the current docs? Seems least dramatic?

Thinking something like:

```python
        if self.default_currency and self.default_currency != DEFAULT_CURRENCY:
            kwargs['default_currency'] = str(self.default_currency)
```

**Comment by Stranger6667** [comment]
Yep, it is a bug - let's start with a simple fix. There are some confusing points that we can clarify/eliminate as the next steps to `1.0` release.

1. `default` vs `default_currency` #474 ;
2. Possibility to use `float` as the default value (I think we need to forbid it);
3. Multiple formats for the `MoneyField`. In general, I think that we need to be stricter. Now we have - `"10 EUR"`, `(10, "EUR")`, `Money(10, "EUR")` or numerics;
4. `moneyed.Money` vs `djmoney.money.Money`, the first one doesn't support `F` expressions + some arithmetics and exchange conversions. Also, there is `MoneyPatched` compatibility class that is deprecated;

We have some extra logic to validate the length of the passed tuple, string, etc. And I'm not sure if there is a real use case for strings and tuples/lists as default values.

I think it would be nice to have some restrictions on our side, that would prevent possible bugs or precision loss on the user side - `float` defaults for example.

**Comment by washeck** [comment]
@Stranger6667 Absolutely agreed with your points. 

Would like to have a milestone or tracking issue for changes planned for 1.0? For our usage it is critical to avoid any magical defaults ('XYZ' currency etc) so I'd be happy to see that included (e.g. ability to set `DEFAULT_CURRENCY=None` in settings).

I can devote some of my working time to improvements of `django-money` but your guidance would be needed. Btw, if you attend Prague's PyVo meetup we can meet there (maybe even today :)

**Comment by benjaoming** [comment]
> Would like to have a milestone or tracking issue for changes planned for 1.0?

That's already open :) We just need to use it.... https://github.com/django-money/django-money/milestone/7

The current `master` branch is targeted for 1.0, and other fixes go to the `0.14.x` branch. You are most welcome to make a suggestion to fix the immediate issue in `0.14.x`, @washeck, and yeah maybe you can find @Stranger6667 in Prague :)

**Comment by Stranger6667** [comment]
Hej!
I'm out of Prague till the end of the next week :) Will be fully available from June and I will be happy to assist with `django-money` even before I'm back :)

**Comment by Stranger6667** [comment]
resolved via #491

I'll create extra issues for the points I mentioned above
